### PR TITLE
ask user whether to continue listening in background

### LINKED ISSUE #141: remove the 'x' button and the 'console' tab from the Cloud Debug UI
1)The cloud debug tool window includes a 'console' tab next to the 'debugger' tab which I don't think displays anything. Ideally it would be removed.

2) in the left margin of the cloud debug tool panel there is a red 'x' button. Clicking it brings up a dialog that says 'do you want to disconnect from the process 'cloud debug'? with the option to terminate the process after disconnect.  I'm not sure what this means in the context of the cloud debugger as their is no local process. Ideally this dialog would not be displayed or the 'x' button would be removed.

3) Also the blue square button that represents stop is kind of confusing.  For some reason it did not register with me as a stop button. I'd be tempted to make it red like the other debugger stop buttons.

### PR DESCRIPTION
**Pull request by elharo**
@patflynn @loosebazooka close #141

### CODE REVIEW

File: core-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java [hunk 1]
```diff
@@ -73,13 +73,23 @@ public void run() {
...
-  private static List<CloudDebugProcessState> getStates() {
+  public synchronized void stopBackgroundListening() {
+    if (myWatchTimer != null) {
+      myWatchTimer.cancel();
+      myWatchTimer = null;
+    }
+  }
+
+
+    private static List<CloudDebugProcessState> getStates() {
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: is the indent off here?

File: core-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java [hunk 2]
```diff
@@ -467,8 +468,10 @@ public SaveAndExitAction() {
     }
 
     @Override
-    public void actionPerformed(AnActionEvent e) {
-      ActionManager.getInstance().getAction(IdeActions.ACTION_STOP_PROGRAM).actionPerformed(e);
+    public void actionPerformed(AnActionEvent event) {
+      ExitDialog exitDialog = new ExitDialog(getXDebugSession().getProject());
+      exitDialog.showAndGetOk();
+      ActionManager.getInstance().getAction(IdeActions.ACTION_STOP_PROGRAM).actionPerformed(event);
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java]: could you add usage tracking here?

**Comment by patflynn** [comment]
Should we be removing the run configuration listening setting as part of this?

File: core-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java (hunk 2 above)
Comment (elharo) [inline core-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java]: done

**Comment by elharo** [comment]
done

File: core-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java (hunk 1 above)
Comment (elharo) [inline core-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java]: done

**Comment by elharo** [comment]
Regarding "Should we be removing the run configuration listening setting as part of this?" I added this, but now I'm not so. If I read it right this setting controls whether we notify when a new snapshot is received both when we're listening and when we're not. If so it still applies.

If I'm wrong about that, then yes, we should take it out.

**Comment by elharo** [comment]
PTAL. Does not yet close the debugger tab. This PR is getting big, so I'm goign to file a sepaarte bug for that.

**Comment by patflynn** [comment]
looks good. Is there a way we could track close and keep watching vs close and stop watching?

**Comment by elharo** [comment]
With Gogole Analytics? sure. we just add

          UsageTrackerService.getInstance()
              .trackEvent(GctTracking.CATEGORY, GctTracking.CLOUD_DEBUGGER,
"close.stoplistening", null);

and

          UsageTrackerService.getInstance()
              .trackEvent(GctTracking.CATEGORY, GctTracking.CLOUD_DEBUGGER,
"close.continuelistening", null);

at the relevant locations. Do we have any particular standards for
labelling these events?




On Fri, Nov 6, 2015 at 3:32 PM, Patrick Flynn <notifications@github.com>
wrote:

> looks good. Is there a way we could track close and keep watching vs close
> and stop watching?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/192#issuecomment-154524234>
> .
>



-- 
Elliotte Rusty Harold
elharo@macfaq.com

**Comment by elharo** [comment]
ping

File: core-plugin/src/com/google/gct/idea/debugger/ui/ExitDialog.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  private final JLabel myLabel = new JLabel(GctBundle.getString("clouddebug.continuelistening"));
+
+  public ExitDialog(@Nullable Project project) {
+    super(project, false, IdeModalityType.MODELESS);
+
+    init();
+
+    setOKButtonText(GctBundle.getString("clouddebug.continue"));
+    setCancelButtonText(GctBundle.getString("clouddebug.stoplistening"));
+  }
```
Comment (patflynn) [inline core-plugin/src/com/google/gct/idea/debugger/ui/ExitDialog.java]: can you add the tracking to differentiate when a user opts to stop vs continue listening?

**Comment by elharo** [comment]
are the labels I suggested above OK or is there some other convention we use?

**Comment by patflynn** [comment]
labels work for me. No conventions as of now that I'm aware of.

**Comment by patflynn** [comment]
+nabila

**Comment by elharo** [comment]
done

**Comment by patflynn** [comment]
LGTM

**Pull request merged**